test/parallel/test-url-parse-invalid-input.js

```diff
@@ -13,8 +13,8 @@ const url = require('url');
   0,
   [],
   {}
-].forEach(function(val) {
-  assert.throws(function() { url.parse(val); }, TypeError);
+].forEach((val) => {
+  assert.throws(() => { url.parse(val); }, /^TypeError: Parameter "url" must be a string, not undefined|boolean|number|object$/);
```

Comment:
I think you want to group `undefined|boolean|number|object` in parentheses, right?
And could you also wrap at the `,` (align the function and the regexp vertically)? We try to keep lines under 80 characters – for regexps that’s not always possible, but it would still be nice to try and keep lines short :)